src/crypto/crypto_util.cc

```diff
@@ -105,6 +103,25 @@ int NoPasswordCallback(char* buf, int size, int rwflag, void* u) {
   return 0;
 }
 
+bool ProcessFipsOptions() {
+  /* Override FIPS settings in cnf file, if needed. */
+  if (per_process::cli_options->enable_fips_crypto ||
+      per_process::cli_options->force_fips_crypto) {
+#if OPENSSL_VERSION_MAJOR >= 3
+    OSSL_PROVIDER* fips_provider = OSSL_PROVIDER_load(nullptr, "fips");
+    if (fips_provider == nullptr)
+      return false;
+    OSSL_PROVIDER_unload(fips_provider);
+
+    return EVP_default_properties_enable_fips(nullptr, 1) &&
+           EVP_default_properties_is_fips_enabled(nullptr);
+#else
+    return 0 == FIPS_mode() && FIPS_mode_set(1);
```

Comment:
At least, in the JS yoda is not allowed:
```suggestion
    return FIPS_mode() == 0 && FIPS_mode_set(1);
```

Reply:
I've updated this now, thanks